### ISSUE TITLE
On Mac M1, importing  '_get_thread_id' from 'numba.np.ufunc.parallel' throws an error. Downgrading to 0.55.2 fixes the problem

### ISSUE THREAD

**Issue by josh-ashkinaze** [numba/numba#8383]
This is causing other packages to error out, which use numba. 

```
import numba
from numb.np.ufunc.parallel import _get_thread_id
```

`ImportError: cannot import name '_get_thread_id' from 'numba.np.ufunc.parallel' (/Users/jashkina/Library/Python/3.8/lib/python/site-packages/numba/np/ufunc/parallel.py)`

It is fixed when downgrading to `0.55.2`, so it must be something in this release which is breaking.

**Comment by gmarkall** [comment]
Thanks for the report - can you outline why this should work please? Was it documented / published as an API for external use somewhere?

**Comment by stuartarchibald** [comment]
For reference, in 0.56 Numba removed the private `numba.np.ufunc.parallel._get_thread_id` function:
https://github.com/numba/numba/blob/2298ad6186d177f39c564046890263b0f1c74ecc/numba/np/ufunc/parallel.py#L652-L659
and replaced it with a similar, but public, API function `numba.get_thread_id`:
https://github.com/numba/numba/blob/f75c45a8dd7c7acfdfa8b41bea18695a88447c3e/numba/np/ufunc/parallel.py#L683-L693

I just noticed that the docs/docstring for this are not very well formatted, suggest fixing for 0.56.1.

**Comment by stuartarchibald** [comment]
Docs update in: #8386.

**Comment by stuartarchibald** [comment]
Closing this as fixed. The API in use is now public, fix is to update to Numba 0.56+. Thanks!
